tests/unit_tests/sparse_blas/include/sparse_reference.hpp

```diff
@@ -232,4 +266,32 @@ void prepare_reference_gemm_data(const intType *ia, const intType *ja, const fpT
     }
 }
 
+template <typename fpType, typename intType>
+void prepare_reference_trsv_data(const intType *ia, const intType *ja, const fpType *a, intType m,
+                                 intType a_ind, oneapi::mkl::uplo uplo_val,
+                                 oneapi::mkl::transpose opA, oneapi::mkl::diag diag_val,
+                                 const fpType *x, fpType *y_ref) {
+    std::size_t mu = static_cast<std::size_t>(m);
+    auto dense_a = sparse_to_dense(ia, ja, a, mu, mu, a_ind, opA, diag_val);
```

Comment:
As I understand, `trsv` solves a system of linear equations. I'm not totally sure what you're doing here, but wouldn't the most simple solution be to generate the x to solve via matrix multiplication first from a generated y?

On host do a matrix multiply with a known *y*:
$$x = A y$$
and then use trsv to try and recover the *y* value.

Reply:
As you say it probably isn't worth changing now. If we find in the future that we have trouble with numerical accuracy, it might be a solution.